app/api/helpers/storage.py

```diff
@@ -162,7 +164,15 @@ def upload_local(uploaded_file, key, **kwargs):
         return get_settings()['static_domain'] + \
             file_relative_path.replace('/static', '')
     url = urlparse(request.url)
-    return url.scheme + '://' + url.hostname + file_relative_path
+
+    # No need to specify scheme-corresponding port
+    port = url.port
+    if port and url.scheme == SCHEMES.get(url.port, None):
+        port = None
+
+    return '{scheme}://{hostname}:{port}{file_relative_path}'.format(
+        scheme=url.scheme, hostname=url.hostname, port=port,
+        file_relative_path=file_relative_path).replace(':None', '')
```

Comment:
Looks good, if you can add tests for this logic, it'd be great

Reply:
Create 3-4 new ones testing different types of URL, you'll have to move this logic in a different function for isolated testing

Reply:
@iamareebjamal Just curious, shouldn't test methods `test_upload_image_local()`, `test_upload_single_file()` and `test_upload_multiple_file()` be in a different test file, say `test_storage.py`?

Reply:
@iamareebjamal How do I simulate requests going to different ports during tests?

Reply:
Okay, I thought so because the url is generated via the `request.url` call within this function.

Reply:
That's the point of Unit Testing. Isolate components from real-world dependencies so that they can be tested for each and every use case. Unit tests use mock arguments to test the correctness of a function so that they can check it is working as expected without simulating real world constraints

Reply:
Thanks, here's what I've come up with so far:
```
def test_upload_local(self):
    upfile = UploadedFile('/path/to/image.png', 'image.png')
    with app.test_request_context():
        upload_url = upload_local(upfile, key='test')
    ...
```
The value of `upload_url` at this point is `http://localhost:5000/static/media/test/clZCY2xkNz/image.png`. Now I want to have different values of port so that I can check the match between expected and actual urls for different ports and schemes. I'm working on tuning the port parameter.